x-pack/plugin/ql/src/test/java/org/elasticsearch/xpack/ql/expression/AttributeMapTests.java

```diff
@@ -37,6 +38,47 @@ private static Attribute a(String name) {
         return new AttributeMap<>(map);
     }
 
+    public void testAttributeMapWithSameAliasesCanResolveAttributes() {
+        Alias param1 = createIntParameterAlias(1, 100);
+        Alias param2 = createIntParameterAlias(2, 100);
+        assertTrue(param1.equals(param2));
+        assertTrue(param1.semanticEquals(param2));
+        // equality on literals
+        assertTrue(param1.child().equals(param2.child()));
+        assertTrue(param1.child().semanticEquals(param2.child()));
+        assertTrue(param1.toAttribute().equals(param2.toAttribute()));
+        assertFalse(param1.toAttribute().semanticEquals(param2.toAttribute()));
+
+        Map<Attribute, Expression> collectRefs = new LinkedHashMap<>();
+        for (Alias a : List.of(param1, param2)) {
+            collectRefs.put(a.toAttribute(), a.child());
+        }
+        // we can look up the same item by both attributes
+        assertNotNull(collectRefs.get(param1.toAttribute()));
+        assertNotNull(collectRefs.get(param2.toAttribute()));
+        AttributeMap<Expression> attributeMap = new AttributeMap<>(collectRefs);
+
+        // validate that all Alias can be e
```

Comment:
nit: "~ wrapper collided" or some other comment correction (can be the next PR, obv.)